src/main/java/com/github/alexcojocaru/mojo/elasticsearch/v2/step/resolveartifact/ElasticsearchArtifactResolverMaster.java

```diff
@@ -148,7 +154,14 @@ private File downloadArtifact() throws IOException
                             : config.getDownloadUrl());
 
         config.getLog().info("Downloading " + downloadUrl + " to " + tempFile);
-        FileUtils.copyURLToFile(downloadUrl, tempFile);
+        URLConnection connection = downloadUrl.openConnection();
+        Optional.ofNullable(config.getDownloadUrlUsername()).ifPresent(username -> {
+            String basicAuthenticationEncoded = Base64.getEncoder().encodeToString((config.getDownloadUrlUsername() + ":" + config.getDownloadUrlPassword()).getBytes(StandardCharsets.UTF_8));
```

Comment:
This would work well only if the password is provided along with the username, in other words both or none. Can you add some validation for that in `buildClusterConfiguration` ?

Reply:
I've added a precondition check and corresponding unit tests